solr/core/src/java/org/apache/solr/storage/SizeAwareDirectory.java

```diff
@@ -91,6 +117,27 @@ public long fileLength(String name) throws IOException {
     }
   }
 
+  @Override
+  public long onDiskFileLength(String name) throws IOException {
+    Sizes ret = fileSizeMap.get(name);
+    SizeAccountingIndexOutput live;
+    if (ret != null) {
+      return ret.onDiskSize;
+    } else if ((live = liveOutputs.get(name)) != null) {
+      if (live.backing instanceof CompressingDirectory.SizeReportingIndexOutput) {
+        return ((CompressingDirectory.SizeReportingIndexOutput) live.backing).getBytesWritten();
+      }
+      // backing IndexOutput does not allow us to get onDiskSize
+      return 0;
```

Comment:
gah -- this doesn't matter for us practically, but I think we may need a conditional here. We should return 0 iff backing IndexOutput `instanceof SizeReportingIndexOutput` _AND_ backing _directory_ `instanceof OnDiskSizeDirectory`. Otherwise, in keeping with how non-OnDiskSizeDirectory onDiskSize is handled, we should indeed fall back to returning `sizeOf` (as before).